src/main/java/uk/gov/hmcts/reform/em/stitching/pdf/PDFMerger.java

```diff
@@ -43,7 +46,11 @@ public File merge() throws IOException {
                 currentPageNumber += tableOfContents.getNumberPages();
             }
 
-            addContainer(bundle);
+            pdfOutlineService = new PDFOutlineService(document, bundle);
```

Comment:
How come this version of the constructor takes some arguments? I am suspicious now

Reply:
I believe I can clean this up. For the reason of two constructor, the first constructor was created with two arguments, 1) document as the creation of the outline was done on it, 2) the bundle for checks though, I believe I can make it with one constructor with only the document as an argument.

Reply:
Yeah, I need to sit down and look at this more thoroughly. I hope we can simplify that class. I am thinking that a single instance should be create at the beginning of the stitching and then the coversheets and pdf merger modify it as they go.

Reply:
I've read through the rest of the code. I it would be better to have the pdfOutlineService.createOutline method return an object PdfOutline. This class could then encapsulate the PDOutlineItem state so that you don't have to track it all throughout this class.

I would say the outlineService should be a dependency for the PDFMerger, it should create the PdfOutline in it's public method and pass it to the constructor of the StatefulPDFMerger